redisraft.h

```diff
@@ -553,6 +553,9 @@ typedef struct SnapshotResult {
     char err[256];
 } SnapshotResult;
 
+extern char * join_type;
+extern char * link_type;
+
```

Comment:
I think there should be a better way to handle this than introduce these globals. If we're trying to keep this reusable and generic, maybe we should just have a callback function for unrecovarable error handling - which will be responsible for producing the apropriate error reply as well perform any other specific cleanup tasks we may need in the future.

Reply:
ok, can get rid of the globals, I'd rather not add another callback yet if not understood what it really adds over the existign code.  I'll just add allocated data to state object and free in its normal callback